src/ResourceManager/Sql/Commands.Sql/ManagedDatabase/Cmdlet/GetAzureSqlManagedDatabase.cs

```diff
@@ -0,0 +1,129 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.ManagedDatabase.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedDatabase.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlManagedDatabase", ConfirmImpact = ConfirmImpact.None)]
+    public class GetAzureSqlManagedDatabase : AzureSqlManagedDatabaseCmdletBase<IEnumerable<AzureSqlManagedDatabaseModel>>
+    {
+        protected const string GetByNameAndResourceGroupParameterSet =
+            "GetManagedDatabaseFromInputParameters";
+
+        protected const string GetByResourceIdParameterSet =
+            "GetManagedDatabaseFromAzureResourceId";
+
+        /// <summary>
+        /// Gets or sets the name of the managed database to use.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The name of the Azure SQL Managed Database to retrieve.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedDatabaseName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure Sql Managed Instance
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "The Azure Sql Managed Instance name.")]
+        [ValidateNotNullOrEmpty]
+        public override string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource id of the Managed database to get
+        /// </summary>
+        [Parameter(ParameterSetName = GetByResourceIdParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource id of Managed Database object to get")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
```

Comment:
In addition to ResourceId, you should have an inputobject parameter set that accepts a ManagedInstance object (name the parameter ManagedInstanceObject).  This parameter set and the ResourceId parameter set should both accept ManagedDatabaseName as an optional parameter name.

When this is done, don't forget to regen help so this is reflected.